### PR TITLE
Allow querying specific releases

### PR DESCRIPTION
**Pull request by karmadolkar**
Ref #4077 

Signed-off-by: Karma Dolkar <karmadolkar29@gmail.com>

### CODE REVIEW

File: bodhi/server/services/graphql.py [hunk 1]
```diff
@@ -48,11 +48,107 @@ class Query(graphene.ObjectType):
     """Allow querying objects."""
 
     allReleases = graphene.List(Release)
+    specificReleases = graphene.Field(lambda: graphene.List(Release),
```
Comment (cverna) [inline bodhi/server/services/graphql.py]: Let's just use the following fields `name, id_prefix, state, composed_by_bodhi`

File: bodhi/server/services/graphql.py [hunk 2]
```diff
@@ -48,11 +48,107 @@ class Query(graphene.ObjectType):
...
     def resolve_allReleases(self, info):
         """Answer Queries by fetching data from the Schema."""
         query = Release.get_query(info)  # SQLAlchemy query
         return query.all()
 
+    def resolve_specificReleases(self, info, **args):
+        """Resolving releases with a given argument"""
+        query = Release.get_query(info)
+
+        id_prefix = args.get("id_prefix")
```
Comment (cverna) [inline bodhi/server/services/graphql.py:65]: same here we can really focus on the following fields `name, id_prefix, state, composed_by_bodhi`

File: bodhi/server/services/graphql.py [hunk 3]
```diff
@@ -48,11 +48,37 @@ class Query(graphene.ObjectType):
     """Allow querying objects."""
 
     allReleases = graphene.List(Release)
+    specificReleases = graphene.Field(
```
Comment (cverna) [inline bodhi/server/services/graphql.py]: I am not a big fan of `specificReleases` for the name of the query, it is a bit long to type for me :P. @StephenCoady do you have an idea on how we could name that ?

File: bodhi/tests/server/services/test_graphql.py [hunk 4]
```diff
@@ -92,3 +92,63 @@ def test_enumfields(self):
...
+            "data": {
+                "specificReleases": [{
+                    "name": "F17"
+                }, {
+                    "name": "F22"
+                }]
+            }
+        }
+
+        executed = client.execute("""{  specificReleases(state: "current"){  name  }}""")
```
Comment (cverna) [inline bodhi/tests/server/services/test_graphql.py]: I would also add a test case where we have multiple attributes something like 
```suggestion
        executed = client.execute("""{  specificReleases(state: "pending", composedByBodhi: false){  name  }}""")
```

**Review by cverna — COMMENTED** [review]
Looking good just a few comments :-)

File: bodhi/server/services/graphql.py (hunk 3 above)
Comment (StephenCoady) [reply bodhi/server/services/graphql.py]: Yeah I'd agree. Lets make it descriptive but simple, how about `getRelease` ?

File: bodhi/tests/server/services/test_graphql.py [hunk 5]
```diff
@@ -92,3 +92,64 @@ def test_enumfields(self):
                 }]
             }
         }
+
+    def test_getRelease(self):
+        """Testing getRelease."""
+        client = Client(schema)
```
Comment (cverna) [inline bodhi/tests/server/services/test_graphql.py:70]: We could use a setup_method in the TestGraphQLService class to declare the clien and release variable (see https://github.com/fedora-infra/bodhi/blob/develop/bodhi/tests/server/test_config.py#L55). That way you don't have to copy/paste that to every tests and you can use self.client and self.release.

**Review by cverna — APPROVED** [review]
LGTM

**Review by StephenCoady — APPROVED** [review]
sorry for the delay reviewing. yeah this looks good to me. I've tried it out too and its working as expected :) well done.

**Pull request merged**